### PR TITLE
ci: Cleanup to maximize available build space

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

## Changelog

- Improvement

## Related Issues

Fixes #issue

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: .github/workflows/developing.yml [hunk 1]
```diff
@@ -22,6 +22,9 @@ jobs:
   check:
     runs-on: ubuntu-latest
     steps:
+      - name: Maximize build space
+        uses: easimon/maximize-build-space@master
```
Comment (Xuanwo) [inline .github/workflows/developing.yml:26]: `easimon/maximize-build-space` supports to set `remove-dotnet: 'true'`, we can replace the original `rm /usr/share/dotnet` with this action in every job.

Comment (Xuanwo) [reply .github/workflows/developing.yml:26]: Also, use `easimon/maximize-build-space@v4` instead to avoid unexpected upstream changes.

**Comment by Xuanwo** [comment]
How about remove all occurence about: 

https://github.com/datafuselabs/databend/blob/1b6bf394d78931f61d46719f4abace5e938d7e46/.github/actions/check/action.yml#L10-L12

**Comment by BohuTANG** [comment]
Emm, `check
You are running out of disk space. The runner will stop working when the machine runs out of disk space. Free space left: 48 MB`

**Comment by Xuanwo** [comment]
Every job is running in the different VM, so I think we need to insert this action before every `actions/checkout@v2`.

BTW, we can free more space via the following args https://github.com/easimon/maximize-build-space/blob/b4d02c14493a9653fe7af06cc89ca5298071c66e/action.yml#L37-L48

And I don't know why the CI failed for out of space, maybe we need to reserve more space for root fs?

After this action:

```
Filesystem                   Size  Used Avail Use% Mounted on
/dev/root                     84G   83G  1.0G  99% /
devtmpfs                     3.4G     0  3.4G   0% /dev
tmpfs                        3.4G  4.0K  3.4G   1% /dev/shm
tmpfs                        696M  1.1M  695M   1% /run
tmpfs                        5.0M     0  5.0M   0% /run/lock
tmpfs                        3.4G     0  3.4G   0% /sys/fs/cgroup
/dev/loop0                    68M   68M     0 100% /snap/lxd/21835
/dev/loop1                    43M   43M     0 100% /snap/snapd/14066
/dev/loop2                    62M   62M     0 100% /snap/core20/1242
/dev/sda15                   105M  5.2M  100M   5% /boot/efi
/dev/sdb1                     14G   13G  100M 100% /mnt
/dev/mapper/buildvg-buildlv   43G   53M   43G   1% /home/runner/work/databend/databend
```

**Comment by Xuanwo** [comment]
Well, the action is too hack for us, how about implement a simple clean action in which we do job like:

https://github.com/easimon/maximize-build-space/blob/master/action.yml#L96-L106

```
echo "Removing unwanted software... "
if [[ ${{ inputs.remove-dotnet }} == 'true' ]]; then
  sudo rm -rf /usr/share/dotnet
fi
if [[ ${{ inputs.remove-android }} == 'true' ]]; then
  sudo rm -rf /usr/local/lib/android
fi
if [[ ${{ inputs.remove-haskell }} == 'true' ]]; then
  sudo rm -rf /opt/ghc
fi
echo "... done"
```

Based of the comment, we can save ~17 + ~11 + ~2.7 = ~31GB.

**Comment by BohuTANG** [comment]
@mergify update

**Comment by Xuanwo** [comment]
After cleanup, we have:

```
Available storage:
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        84G   34G   50G  41% /
devtmpfs        3.4G     0  3.4G   0% /dev
tmpfs           3.4G  4.0K  3.4G   1% /dev/shm
tmpfs           696M  1.1M  695M   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           3.4G     0  3.4G   0% /sys/fs/cgroup
/dev/loop0       62M   62M     0 100% /snap/core20/1242
/dev/loop1       68M   68M     0 100% /snap/lxd/21835
/dev/sda15      105M  5.2M  100M   5% /boot/efi
/dev/loop2       43M   43M     0 100% /snap/snapd/14066
/dev/sdb1        14G  4.1G  9.0G  32% /mnt
```

**Comment by Xuanwo** [comment]
@BohuTANG @PsiACE @sundy-li I think this PR is ready to go!

**Pull request merged**